src/jvmMain/kotlin/io/github/vyfor/kpresence/ipc/Connection.kt

```diff
@@ -91,16 +95,20 @@ actual class Connection {
     override fun write(opcode: Int, data: String?) {
       pipe?.let { stream ->
         try {
-          val bytes = data?.encodeToByteArray()
-          val buffer = ByteArray((bytes?.size ?: 0) + 8)
+          try {
+            val bytes = data?.encodeToByteArray()
+            val buffer = ByteArray((bytes?.size ?: 0) + 8)
 
-          buffer.putInt(opcode.reverseBytes())
-          if (bytes != null) {
-            buffer.putInt(bytes.size.reverseBytes(), 4)
-            bytes.copyInto(buffer, 8)
-          }
+            buffer.putInt(opcode.reverseBytes())
+            if (bytes != null) {
+              buffer.putInt(bytes.size.reverseBytes(), 4)
+              bytes.copyInto(buffer, 8)
+            }
 
-          stream.write(ByteBuffer.wrap(buffer), 0).get()
+            stream.write(ByteBuffer.wrap(buffer), 0).get()
+          } catch (e: ExecutionException) {
+            throw (e.cause ?: throw PipeReadException(e.message.orEmpty()))
+          }
```

Comment:
Please do the same here, and also:
```suggestion
          } catch (e: ExecutionException) {
            throw (e.cause ?: throw PipeWriteException(e.message.orEmpty()))
          }
```